pkg/apis/hco/v1alpha1/hyperconverged_types.go

```diff
@@ -69,3 +73,43 @@ type HyperConvergedList struct {
 func init() {
 	SchemeBuilder.Register(&HyperConverged{}, &HyperConvergedList{})
 }
+
+type KubevirtConfigurations struct {
+	ResourceVersion                   string            `json:"ResourceVersion,omitempty"`
+	UseEmulation                      bool              `json:"UseEmulation,omitempty"`
+	MigrationConfig                   *MigrationConfig  `json:"MigrationConfig,omitempty"`
+	ImagePullPolicy                   corev1.PullPolicy `json:"ImagePullPolicy,omitempty"`
+	MachineType                       string            `json:"MachineType,omitempty"`
+	CPUModel                          string            `json:"CPUModel,omitempty"`
+	CPURequest                        resource.Quantity `json:"CPURequest,omitempty"`
+	MemoryOvercommit                  int               `json:"MemoryOvercommit,omitempty"`
+	EmulatedMachines                  []string          `json:"EmulatedMachines,omitempty"`
+	FeatureGates                      *FeatureGates     `json:"FeatureGates,omitempty"`
+	LessPVCSpaceToleration            int               `json:"LessPVCSpaceToleration,omitempty"`
+	NodeSelectors                     map[string]string `json:"NodeSelectors,omitempty"`
+	NetworkInterface                  string            `json:"NetworkInterface,omitempty"`
+	PermitSlirpInterface              bool              `json:"PermitSlirpInterface,omitempty"`
+	PermitBridgeInterfaceOnPodNetwork bool              `json:"PermitBridgeInterfaceOnPodNetwork,omitempty"`
+	SmbiosConfig                      *cmdv1.SMBios     `json:"SmbiosConfig,omitempty"`
+}
+
+type MigrationConfig struct {
+	ParallelOutboundMigrationsPerNode *uint32            `json:"parallelOutboundMigrationsPerNode,omitempty"`
+	ParallelMigrationsPerCluster      *uint32            `json:"parallelMigrationsPerCluster,omitempty"`
+	BandwidthPerMigration             *resource.Quantity `json:"bandwidthPerMigration,omitempty"`
+	NodeDrainTaintKey                 *string            `json:"nodeDrainTaintKey,omitempty"`
+	ProgressTimeout                   *int64             `json:"progressTimeout,omitempty"`
+	CompletionTimeoutPerGiB           *int64             `json:"completionTimeoutPerGiB,omitempty"`
+	UnsafeMigrationOverride           bool               `json:"unsafeMigrationOverride"`
+	AllowAutoConverge                 bool               `json:"allowAutoConverge"`
+}
+
+// FeatureGates represents feature-gates to be enabled or disabled in kubevirt configMap
+type FeatureGates struct {
```

Comment:
This seems to not include all feature gates (e.g. Ignition is not there). An array might be more on how it is done in k8s when it comes to feature gates.